src/server/scripts/EasternKingdoms/ScarletEnclave/chapter2.cpp

```diff
@@ -15,214 +15,298 @@
+    POINT_ID_6                  = 6,
+    POINT_ID_10                 = 10,
+
+    SUMMON_ACOLYTES_0           = 0,
+    SUMMON_ACOLYTES_1           = 1,
+    SUMMON_ACOLYTES_2           = 2,
+    SUMMON_VALROTH              = 3,
+
+    QUEST_BLOODY_BREAKOUT       = 12727,
+
+    NPC_FAKE_VALROTH            = 29011,
+    NPC_VALROTH                 = 29001,
+    NPC_ACOLYTE                 = 29007,
+    NPC_KOLTIRA                 = 28912,
+    NPC_KOLTIRA_MOUNT           = 25445,
+
+    SPELL_KOLTIRA_TRANSFORM     = 52899,
+    SPELL_ANTI_MAGIC_ZONE       = 52894,
+    SPELL_HERO_AGGRO            = 53628
+};
 
-    struct npc_koltira_deathweaverAI : public EscortAI
+Position const koltiraPos[3] =
+{
+        { 1653.36f, -6038.34f, 127.584f },
+        { 1653.765f, -6035.075f, 127.5844f },
+        { 1651.89f, -6037.101f, 127.5844f }
+};
+
+struct npc_koltira_deathweaver : public ScriptedAI
+{
+    npc_koltira_deathweaver(Creature* creature) : ScriptedAI(creature), _summons(me)
     {
-        npc_koltira_deathweaverAI(Creature* creature) : EscortAI(creature)
-        {
-            Initialize();
-            me->SetReactState(REACT_DEFENSIVE);
-        }
+    }
 
-        void Initialize()
-        {
-            wave = 0;
-            waveTimer = 3000;
-            valrothGUID.Clear();
-        }
+    bool GossipHello(Player* player) override
+    {
+        ObjectGuid const guid = me->GetGUID();
+        _playerGUID = player->GetGUID();
 
-        void Reset() override
+        if (me->IsQuestGiver())
+            player->PrepareQuestMenu(guid);
+
+        // override default gossip
+        if (_eventGossip)
         {
-            if (!HasEscortState(STATE_ESCORT_ESCORTING))
-            {
-                Initialize();
-                me->LoadEquipment(0, true);
-                me->RemoveAurasDueToSpell(SPELL_ANTI_MAGIC_ZONE);
-                me->RemoveAurasDueToSpell(SPELL_KOLTIRA_TRANSFORM);
-                me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
-            }
+            SendGossipMenuFor(player, TEXT_ID_EVENT, guid);
+            return true;
         }
 
-        void WaypointReached(uint32 waypointId, uint32 /*pathId*/) override
+        return false;
+    }
+
+    void QuestAccept(Player* /* player */, Quest const* quest) override
+    {
+        if (quest->GetQuestId() == QUEST_BLOODY_BREAKOUT)
+            _events.ScheduleEvent(EVENT_INTRO_0, 500ms);
+    }
+
+    void Reset() override
+    {
+        if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+            player->FailQuest(QUEST_BLOODY_BREAKOUT);
+
+        me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_NPC);
+        me->SetFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+        me->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_STAND_STATE, UNIT_STAND_STATE_DEAD);
+
+        me->RemoveAurasDueToSpell(SPELL_ANTI_MAGIC_ZONE);
+        me->RemoveAurasDueToSpell(SPELL_KOLTIRA_TRANSFORM);
+        DoCastSelf(SPELL_HERO_AGGRO);
```

Comment:
This is wrong the spell 53628 is triggered by the aura 53627 that must be added to Koltira in EVENT_OUTRO_4